src/t_set.c

```diff
@@ -239,7 +240,7 @@ void setTypeConvertA(robj *setobj, int enc, alloc a) {
 
     if (enc == OBJ_ENCODING_HT) {
         int64_t intele;
-        dict *d = dictCreate(&setDictType,NULL);
+        dict *d = dictCreate((a->alloc == z_alloc->alloc ? &setDictTypeZ : &setDictTypeM),NULL);
```

Comment:
I think better solution for this is to add comparator for whole **alloc** structure - You could do this in separate PR - intention of this comment is to mark the place of compare operation occurrence, same for https://github.com/KFilipek/redis/blob/e149bb6a23d0ca112ea0108345eeb9bb4385271e/src/object.c#L1093

Reply:
Okey, I will do this change in separate PR.